pkg/syncer/metadatasyncer.go

```diff
@@ -460,7 +460,7 @@ func pvcUpdated(oldObj, newObj interface{}, metadataSyncer *metadataSyncInformer
 	migrationEnabled := metadataSyncer.coCommonInterface.IsFSSEnabled(ctx, common.CSIMigration)
 	// Verify if csi migration is ON and check if there is any label update or migrated-to annotation was received for the PVC
 	if migrationEnabled && pv.Spec.VsphereVolume != nil {
-		if !migratedToAnnotationExists(ctx, newPvc.Annotations) {
+		if !migratedToAnnotationExists(ctx, newPvc.Annotations) && newPvc.Annotations[common.AnnStorageProvisioner] == common.InTreePluginName {
```

Comment:
When new PVC is created using legacy Storage Class after the migration is enabled, it does not get migrated-to annotation.

PVC will have following annotations

```
               Annotations:   pv.kubernetes.io/bind-completed: yes
               pv.kubernetes.io/bound-by-controller: yes
               volume.beta.kubernetes.io/storage-provisioner: csi.vsphere.vmware.com
```


For new PV created after the migration is enabled, we have the following annotations.

```
             Annotations:     pv.kubernetes.io/provisioned-by: csi.vsphere.vmware.com
```

Considering above we need to change the logic here.